failure of ex. 5: removal of Elem::neighbor()

Dear all, 
after upgrading my branch since quite a while, the unit tests fail for me saying
```
***************************************************************
* Done Running Example miscellaneous_ex5:
*   ./example-opt
***************************************************************
PASS: run.sh
=============
1 test passed
=============
......
...
..
make[3]: Entering directory '/home/tm162/bin/libmesh/libmesh-1.0/examples/miscellaneous/miscellaneous_ex6'
  CXX      example_dbg-miscellaneous_ex6.o
miscellaneous_ex6.C: In function ‘void add_cube_convex_hull_to_mesh(libMesh::MeshBase&, libMesh::Point, libMesh::Point)’:
miscellaneous_ex6.C:280:17: error: ‘class libMesh::Elem’ has no member named ‘neighbor’; did you mean ‘n_neighbors’?
       if (elem->neighbor(s) == nullptr)
                 ^~~~~~~~
                 n_neighbors
miscellaneous_ex6.C:283:46: error: ‘class libMesh::Elem’ has no member named ‘side’; did you mean ‘id’?
           std::unique_ptr<Elem> side = elem->side(s);
                                              ^~~~
                                              id
Makefile:828: recipe for target 'example_dbg-miscellaneous_ex6.o' failed
make[3]: *** [example_dbg-miscellaneous_ex6.o] Error 1
make[3]: Leaving directory '/home/tm162/bin/libmesh/libmesh-1.0/examples/miscellaneous/miscellaneous_ex6'
..
...
.....
```
and indeed, it seems as if the ``Elem::neighbor(unsigned int)``-function is removed!?
I didn't find a respective commit now, but from first glance, these functions indeed seem to be not available anymore.

Are they removed and mysteriously not covered by some tests yet or can anybody help me figuring out what is going on?

Best regards

--- a/examples/miscellaneous/miscellaneous_ex6/miscellaneous_ex6.C
+++ b/examples/miscellaneous/miscellaneous_ex6/miscellaneous_ex6.C
@@ -277,10 +277,10 @@ void add_cube_convex_hull_to_mesh(MeshBase & mesh,
 
   for (auto & elem : cube_mesh.element_ptr_range())
     for (auto s : elem->side_index_range())
-      if (elem->neighbor(s) == nullptr)
+      if (elem->neighbor_ptr(s) == nullptr)
         {
           // Add the node IDs of this side to the set
-          std::unique_ptr<Elem> side = elem->side(s);
+          std::unique_ptr<Elem> side = elem->side_ptr(s);
 
           for (auto n : side->node_index_range())
             node_id_map.insert(std::make_pair(side->node_id(n), /*dummy_value=*/0));

